display_list/dl_builder.cc

```diff
@@ -108,12 +102,13 @@ sk_sp<DisplayList> DisplayListBuilder::Build() {
   save_stack_.pop_back();
   Init(rtree != nullptr);
 
-  storage_.realloc(bytes);
+  storage_.trim();
+  offsets_.shrink_to_fit();
```

Comment:
Also if the DLBuilder is meant to be reset I think you may need to swap in a new vector because I don't think std::move is guaranteed to reset it to a neutral state like it does for pointers

Reply:
We test reusability. I've been wanting to get rid of re-usability for a while now, but I think there is some code somewhere that relied on that.

Reply:
Yeah it would be good to remove it, because then we can have build simply take the memory and leave this object in an invalid state

Reply:
I switched to more deliberate uses of `std::swap` to transfer the data.

Reply:
Removing reusability is a job for a separate PR as it will have potential consequences outside of this part of the code.

Reply:
(There is already an issue filed for that) https://github.com/flutter/flutter/issues/126589